### DISCUSSION TITLE
Examples of Directus as headless CMS

### DISCUSSION THREAD

**Discussion by mattbab** [directus/directus#12940] [Archive]
Directus is often touted as a headless CMS across the web; often being compared to Storyblok and Contentful as alternatives to WordPress and other full stack CMS'. Having experienced working with Storyblok with Nuxt, I seem to be missing some fundamental explanation of how Directus compares. I definitely see how Directus could be used as a tool to make my own headless CMS, but it isn't directly comparable from what I see. Personally, I see it as more of a database management tool (which aligns with its roots) or database-to-api bridge than something that really helps with serving web content.

Is this a market that the Directus team is actively attempting to cover, and if so, can you point to any examples or tutorials where it's successfully been used in such a way?

**Comment by benhaynes** [comment]
Directus is a data platform, but content is just data, and there are many companies (including Fortune 100) using Directus as a headless CMS. If you compare our platform with any of the other Headless CMS out there, you will find ours is objectively more flexible, feature-rich, intuitive, and powerful.

So, instead of me listing a million Headless CMS features we have... I would flip it around: are there any features you think we're missing from the content management side?

**Reply by mattbab** [reply]
Thank you for your response. It's clear to me how I could use the features of Directus to build a headless CMS myself, however, my question was if there are examples or tutorials available that describe using Directus as a headless CMS. If it's marketed as a headless CMS then I would expect to be able to find guides to use it as such. After spending several hours searching for such, I've only found blog posts which describe Directus as a headless CMS, describe the purposes of a headless CMS, and then walkthrough using Directus as a database management tool - focusing on features that I could arguably achieve quicker/simpler with Adminer or pgAdmin - or a BaaS. Neither of which satisfy the requirements for a headless CMS on their own.

What's missing?
  - A starter schema describing a basic content hierarchy. It's not hard to build this myself, but without it, it's a hurdle to overcome before anyone can start seeing it as a CMS - it's advertised purpose.
  - Supported clients for the most popular SSGs. There are community libraries available, many of which have long gone stale. My research has specifically been around using Nuxt (or any Vue SSG) with Directus. Even when searching for Next.js or Gatsby, their tutorials lack direction or actionable steps to achieve a basic integration. 
  - Sample projects, starter templates, or reference implementations to achieve a basic POC.

I understand Directus is an open source project and depends on community contributions which are largely out of Directus' control. However, with the pivot to getting funded by offering hosting/support, it seems reasonable to have onboarding material to support one of the the advertised uses of the application.

I'll restate part of what you said based on my experience with other platforms: Directus is more flexible and powerful than any other headless CMS out there - because it makes no assumptions or forces any opinions about your _data_. But that's at the expense of any guidance what-so-ever.

**Reply by azrikahar** [reply]
For some examples/tutorials, perhaps you can check out [these guides]( https://directus.io/guides) (in case you haven't) in the meantime, and the source code for the examples showcased can be found here: https://github.com/directus/examples. Hope they can be helpful here.

As for the overall missing points, our resident technical content writer @erondpowell definitely has a ton of plans which does involve them (as far as I'm aware of)! However as we can see from all the recent docs related PRs by him, there's a lot of grounds to cover holistically at the moment. I'll let him elaborate further if needed since I neither can nor should encapsulate all the great things he has in the pipeline. I wouldn't be able to do them justice 😄

**Reply by benhaynes** [reply]
Azri beat me to it... those Guides on our website are the main resources for getting started with SSG/SSR frameworks. 👍

**Reply by erondpowell** [reply]
@mattbab Yeah, agreed! guides are super important.
Let me know if those that Azri sent work out for you.
Like he said, we have quite a bit of foundational documentation to catch up on.
However, the plan is to really boost guides/tutorials, detailing every feature and myriad use-cases.
If you have any specific feedback on what you value in guides/tutorials, we'd love to hear your ideas!

**Reply by quentinglorieux** [reply]
Hi there,

I have been working on the same project for a while: making a WordPress clone using directus and Nuxt 3.
I fully agree that documentation is missing and the one example the directus website is by far too small to be useful.
What would be helpful is a demo blog site with say: 
- a few posts
- a archive page
- some relational data exposed 
- a user directory page 
- some singleton page features for the landing page

What i have trouble to find online right now is:

- what is the best way to fetch data with directus and say Nuxt ? Is there a recommended way ? So guidelines ? Some tips to retrieve relational fields ? To display HTML ?

- how to build a landing page with Nuxt and directus (the clever or recommended way ?)

- what to do with users ? If i have a large directory of people in my company that i want to present on my blog, should i use the user collection or should i create an employee collection ? What is the best approach depending on the applications ?

**Reply by quentinglorieux** [reply]
And BTW the Nuxt3 guide screenshot is incorrect and differ from the GitHub code which does work 
(The post title and content is not displayed in the code on the directus website)

**Reply by azrikahar** [reply]
> And BTW the Nuxt3 guide screenshot is incorrect and differ from the GitHub code which does work (The post title and content is not displayed in the code on the directus website)

Would you mind describing what is the difference? By `directus website` and `Nuxt3 guide`, does both of them mean this page: https://directus.io/guides/get-started-building-a-nuxt-3-blog-with-directus/? Or perhaps you were referring to another site/page?
